File: src/main/clients/openrgb/effects/DanceFloor.ts
import { RGBColor } from '@main/clients/openrgb/client/classes/RGBColor';
import { AbstractEffect } from '@main/clients/openrgb/effects/base/AbstractEffect';

class DanceFloor extends AbstractEffect {
  public constructor() {
    super('Dance floor', false);
  }

  private getRandom(length): Array<RGBColor> {
    const random: Array<RGBColor> = [];
    for (let i = 0; i < length; ++i) {
      random.push(RGBColor.fromHSV(Math.random() * 359, Math.random() * 0.25 + 0.75, 1));
    }

    return random;
  }

  protected async applyEffect(): Promise<void> {
    while (this.isRunning) {
      this.devices.forEach((dev) => {
        this.setColors(dev, this.getRandom(dev.colors.length));
      });
      await this.sleep(500);
    }
    this.hasFinished = true;
  }
}

export const danceFloor = new DanceFloor();

File: lib/clients/openrgb/effects/dance_floor.py
import random

from lib.clients.openrgb.effects.base.abstract_effect import AbstractEffect
from lib.utils.singleton import singleton
from lib.utils.openrgb import OpenRGBUtils


@singleton
class DanceFloor(AbstractEffect):
    """Dance floor effect"""

    def __init__(self):
        super().__init__("Dance floor")

    def _get_random(self, length):
        randoms = []
        for _ in range(length):
            randoms.append(
                OpenRGBUtils.from_hsv(
                    random.random() * 359, random.random() * 0.25 + 0.75, 1
                )
            )
        return randoms

    def apply_effect(self):
        while self.is_running:
            for dev in self.devices:
                self._set_colors(dev, self._get_random(len(dev.colors)))
            self._sleep(0.5)  # 500 ms


dance_floor = DanceFloor()
